service/controller/credential/handler/azureclusteridentity/create.go

```diff
@@ -0,0 +1,184 @@
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	legacySecret, err := key.ToSecret(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.Debugf(ctx, "Found secret %q in namespace %q", legacySecret.Name, legacySecret.Namespace)
+
+	err = r.ensureNewSecret(ctx, legacySecret)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.ensureAzureClusterIdentity(ctx, legacySecret)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *Resource) ensureNewSecret(ctx context.Context, legacySecret corev1.Secret) error {
+	newName := newSecretName(legacySecret)
+	newNamespace := newSecretNamespace(legacySecret)
+
+	r.logger.Debugf(ctx, "Looking for Secret %q in namespace %q", newName, newNamespace)
+
+	clientSecret := string(legacySecret.Data["azure.azureoperator.clientsecret"])
+
+	existing := &corev1.Secret{}
+	err := r.ctrlClient.Get(ctx, client.ObjectKey{Namespace: newNamespace, Name: newName}, existing)
+	if errors.IsNotFound(err) {
+		r.logger.Debugf(ctx, "Secret %q wasn't found in namespace %q, creating it", newName, newNamespace)
+
+		// We need to create the secret.
+		newSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        newName,
+				Namespace:   newNamespace,
+				Labels:      nil,
+				Annotations: nil,
+			},
+			StringData: map[string]string{
+				secretDataFieldName: clientSecret,
+			},
+		}
+
+		err := r.ctrlClient.Create(ctx, newSecret)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.Debugf(ctx, "Secret %q created in namespace %q", newName, newNamespace)
+
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.Debugf(ctx, "Secret %q found in namespace %q", newName, newNamespace)
+
+	currentClientSecret := string(existing.Data[secretDataFieldName])
+	if currentClientSecret != clientSecret {
+		r.logger.Debugf(ctx, "Secret %q is outdated, updating", newName)
+
+		existing.StringData = map[string]string{
+			secretDataFieldName: clientSecret,
+		}
+		err := r.ctrlClient.Update(ctx, existing)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		r.logger.Debugf(ctx, "Secret %q updated successfully", newName)
+		return nil
+	}
+
+	r.logger.Debugf(ctx, "Secret %q is up to date", newName)
+
+	return nil
+}
+
+func (r *Resource) ensureAzureClusterIdentity(ctx context.Context, legacySecret corev1.Secret) error {
+	newName := newSecretName(legacySecret)
+	newNamespace := newSecretNamespace(legacySecret)
+
+	clientID := string(legacySecret.Data["azure.azureoperator.clientid"])
+	tenantID := string(legacySecret.Data["azure.azureoperator.tenantid"])
+
+	desiredSpec := v1alpha3.AzureClusterIdentitySpec{
+		Type:     v1alpha3.ServicePrincipal,
+		ClientID: clientID,
+		ClientSecret: corev1.SecretReference{
+			Name:      newName,
+			Namespace: newNamespace,
+		},
+		TenantID:          tenantID,
+		AllowedNamespaces: make([]string, 0),
```

Comment:
This means the AzureClusterIdentity can be used by any AzureCluster in any namespace.
Should we maybe restrict to the org namespace?